.github/workflows/python.yaml

```diff
@@ -31,17 +29,28 @@ jobs:
     steps:
     - uses: actions/checkout@v3
     - name: Set up Python 3.11
-      uses: actions/setup-python@v3
+      uses: actions/setup-python@v4
       with:
         python-version: 3.11
+        cache: 'pip'
+        cache-dependency-path: |
+          dev-deps.lock
+          pyproject.toml
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip
         if [ -f dev-deps.lock ]; then pip install -r dev-deps.lock; fi
         pip install -e . --no-deps
-    - name: Lint with pre-commit
-      run: |
-        pre-commit run -av
+    - name: Cache linter tool caches
+      uses: actions/cache@v3
+      with:
+        path: |
+          ~/.cache/ruff/
+          ~/.cache/mypy/
+        key: ${{ runner.os }}-${{ hashFiles('dev-deps.lock', 'pyproject.toml', '.pre-commit-config.yaml') }}-linter-cache
```

Comment:
Why `dev-deps.lock` and `pyproject.toml`? For caching on the pre-commit version, too?

Reply:
I'm not really sure about `pyproject.toml` - it only contains a "greater-or-equal" pin for `pre-commit`, and hashing it will cause cache misses on option changes for all tools (and unrelated metadata changes, too). The relevant version data seems to be contained entirely in `dev-deps.lock`.

What do you think?

Reply:
Fine with me, I just like to err on the side of caution when it comes to caching. Will change